core/src/test/java/org/elasticsearch/search/aggregations/InternalAggregationTestCase.java

```diff
@@ -122,4 +136,49 @@ protected final T createUnmappedInstance(String name) {
     protected NamedWriteableRegistry getNamedWriteableRegistry() {
         return namedWriteableRegistry;
     }
+
+    public void testFromXContent() throws IOException {
+        final NamedXContentRegistry xContentRegistry = xContentRegistry();
+        final T aggregation = createTestInstance();
+
+        final ToXContent.Params params = new ToXContent.MapParams(singletonMap(RestSearchAction.TYPED_KEYS_PARAM, "true"));
+        final boolean humanReadable = randomBoolean();
+        final XContentType xContentType = randomFrom(XContentType.values());
+        final BytesReference originalBytes = toShuffledXContent(aggregation, xContentType, params, humanReadable);
+
+        Aggregation parsedAggregation;
+        try (XContentParser parser = xContentType.xContent().createParser(xContentRegistry, originalBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+
+            String currentName = parser.currentName();
+            int i = currentName.indexOf(InternalAggregation.TYPED_KEYS_DELIMITER);
+            String aggType = currentName.substring(0, i);
+            String aggName = currentName.substring(i + 1);
+
+            parsedAggregation = parser.namedObject(Aggregation.class, aggType, aggName);
+
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+
+            final BytesReference parsedBytes = toXContent((ToXContent) parsedAggregation, xContentType, params, humanReadable);
+            assertToXContentEquivalent(originalBytes, parsedBytes, xContentType);
+            assertFromXContent(aggregation, parsedAggregation);
+
+        } catch (ElasticsearchException e) {
+            // TODO Remove this when all aggregations can be parsed back.
```

Comment:
shall we use //norelease instead of TODOs here just to make sure? Though I don't think those fail the build anymore, so maybe it is the same after all. I am a bit paranoid on forgetting these. I would love to merge this branch back with no TODOs left.